package.json

```diff
@@ -59,22 +59,22 @@
     "postcss-simple-vars": "6.0.3",
     "prettier": "2.5.1",
     "resolve": "1.20.0",
-    "rollup-plugin-dts": "5.2.0",
+    "rollup-plugin-dts": "5.3.0",
     "rollup-plugin-hashbang": "2.2.2",
     "strip-json-comments": "4.0.0",
     "svelte": "3.46.4",
     "terser": "^5.16.0",
     "ts-essentials": "9.1.2",
     "tsconfig-paths": "3.12.0",
     "tsup": "6.6.1",
-    "typescript": "4.9.5",
+    "typescript": "5.0.2",
     "vitest": "0.28.4",
     "wait-for-expect": "3.0.2"
   },
   "peerDependencies": {
     "@swc/core": "^1",
     "postcss": "^8.4.12",
-    "typescript": "^4.1.0"
+    "typescript": ">=4.1.0"
```

Comment:
Should this not be something more like `^4.1.0 || ^5.0.0` since a newer version could break this again?

Reply:
in typescript, new minor versions, i.e 4.2 to 4.3 have the same potential breaking changes than major versions, in fact TS 5.0 does not break tsup at all if you don't use some of the new features

Reply:
This is unfortunately true (see microsoft/TypeScript#14116). If TypeScript followed semantic versioning, we would have major version bumps every single release. It is likely that we'd be at TypeScript 50+ by now! To Ryan Cavanaugh, the frequent major version bumps of TypeScript hence [effectively render the semver scheme meaningless](https://github.com/microsoft/TypeScript/issues/14116#issuecomment-280592571).